documentation/src/main/asciidoc/topics/xml/persistence_sql_query.xml

```diff
@@ -4,12 +4,12 @@
                       dialect="POSTGRES"
                       shared="true">
       <queries key-columns="isbn">
-        <select-single>SELECT books.id, books.isbn, title, authors.name, authors.country FROM books JOIN authors ON books.author_id = authors.id WHERE books.id=:id</select-single>
-        <select-all>SELECT books.id, books.isbn, title, authors.name, authors.country FROM books JOIN authors ON books.author_id = authors.id</select-all>
+        <select-single>SELECT isbn, title FROM books WHERE isbn = :key</select-single>
+        <select-all>SELECT isbn, title FROM books</select-all>
         <delete-single>DELETE FROM books WHERE isbn = :key</delete-single>
         <delete-all>DELETE FROM books</delete-all>
         <upsert>INSERT INTO books (isbn, title) VALUES (:key, :value) ON CONFLICT (ispn) DO UPDATE SET title = :value</upsert>
-        <size>SELECT COUNT(*) FROM books JOIN authors ON books.author_id = authors.id</size>
+        <size>SELECT COUNT(*) FROM books</size>
       </queries>
```

Comment:
@wburns Hey again, I'm a bit stuck with this and there are a couple things going on here. 

1. The main issue is that the example is confusing or not detailed enough to show the real use case for SQL query stores. I took this from the design doc at: https://github.com/infinispan/infinispan-designs/blob/main/SQL%20Store.adoc

2. I'm not a SQL guru and took some guesses based on Katia's simple tutorial PR at: https://github.com/infinispan/infinispan-simple-tutorials/pull/150/files#diff-270f6cda87b35fdaa2048f8224b6f1ee5328b9ef1898ba7507bdca2f391aa88fR13

I'm willing to learn but, for the purposes of getting this PR merged, I need to call on some expert guidance. Where can we get an query example that uses a JOIN and does an UPSERT and DELETE and all that good stuff? I think what we want to achieve is something that makes it easier to distinguish from the table store. But maybe I'm going overboard.

If that's the case, we just align docs with what's in the test (and is the same as the design doc): https://github.com/infinispan/infinispan/blob/7b96c0b55616b9a80cef2a9926009b46ee8a2c96/persistence/sql/src/test/resources/configs/all/sql-query.xml#L10

Reply:
My original point is that the variables (ie. start with `:`, e.g. :isbn) have to match the other statement. So a select / select all must return all sql variables (primary key and values), e.g. `title` - note no `:`. The select must take all the primary key fields as arguments `e.g. WHERE isbn = :isbn"`

The joins you had before were quite close, however they seem to be gone now, so I can't give you more specific suggestions.

Reply:
Thanks @wburns I basically took the joins from Katia's tutorial: https://github.com/infinispan/infinispan-simple-tutorials/pull/150/files#diff-270f6cda87b35fdaa2048f8224b6f1ee5328b9ef1898ba7507bdca2f391aa88fR13 

Give me another minute to merge your suggestions and I can add another commit with the joins.